.github/workflows/proto.yml

```diff
@@ -75,3 +75,34 @@ jobs:
           BASE_BRANCH="..origin/${{ steps.setup.outputs.base-branch }}"
           export BASE_BRANCH
           make proto-update-check
+  proto_format_and_gen:
+    runs-on: ubuntu-latest
+    name: Proto Format and Generation Check
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+      - name: Setup Go
+        uses: actions/setup-go@v6
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+      - name: Check Proto Formatting
+        run: |
+          make proto-format
+          if [ -n "$(git status --porcelain)" ]; then
+            echo "::error::Proto files need formatting. Run 'make proto-format' locally."
+            git status --porcelain
+            git diff
+            exit 1
+          fi
+      - name: Check Proto Generation
+        run: |
```

Comment:
Please add `if: always()` to both of these steps so that they'll run even if something before it fails.

We do this to the first one for consistency and so that it's there if we add checks before it, or move the checks around in ordering.